File: tampermonkey/financial_education.js
// ==UserScript==
// @name         OpenAI GPT-3: Financial Education
// @version      1
// @description  Generates a blog article using OpenAI's GPT-3 API and prints the response in the console with elapsed time
// @match        *://*/*
// @grant        none
// ==/UserScript==

(async () => {
    'use strict';

    const getAPIKey = async () => {
        // Obter a chave de API
        const res = await fetch("http://127.0.0.1:2334/OPENAI_API_KEY");
        const data = await res.json();
        return data.data.OPENAI_API_KEY;
    }

    const generateResponse = async (apiKey, prompt, model_engine, max_tokens, temperature) => {
        // Gerar resposta do GPT-3
        const response = await fetch(`https://api.openai.com/v1/engines/${model_engine}/completions`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${apiKey}`
            },
            body: JSON.stringify({
                prompt: prompt,
                max_tokens: max_tokens,
                temperature: temperature
            })
        });
        const data = await response.json();
        return data.choices[0].text;
    }


    // Obter a chave de API
    const apiKey = await getAPIKey();

    // Prompt para gerar o artigo
    const prompt = "escreva um artigo de blog falando sobre educação financeira";
    console.log("prompt: ", prompt);

    // Parâmetros para GPT-3
    const model_engine = "text-davinci-002";
    const max_tokens = 1024;
    const temperature = 0.5;

    const res = await generateResponse(apiKey, prompt, model_engine, max_tokens, temperature);
    console.log(res);
})();

File: financial_education.py
# -*- coding: utf-8 -*-
# ---
# jupyter:
#   jupytext:
#     cell_metadata_filter: -all
#     custom_cell_magics: kql
#     text_representation:
#       extension: .py
#       format_name: percent
#       format_version: '1.3'
#       jupytext_version: 1.11.2
#   kernelspec:
#     display_name: openapi
#     language: python
#     name: python3
# ---

# %%
from dotenv import load_dotenv
import time
import os
import openai


load_dotenv(".env")

# Authenticate with OpenAI API
openai.api_key = os.environ["OPENAI_API_KEY"]

# Prompt for generating article
prompt = "escreva um artigo de blog falando sobre educação financeira"

# Parameters for GPT-3
model_engine = "text-davinci-002"
max_tokens = 1024
temperature = 0.5

# Generate response from GPT-3
start = time.time()
response = openai.Completion.create(engine=model_engine,
                                    prompt=prompt,
                                    max_tokens=max_tokens,
                                    temperature=temperature)
elapsed_time = time.time() - start

# Print response
print(response.choices[0].text)
print(f"\nElapsed Time: {elapsed_time:.2f} seconds")


# %% [markdown]
# A importância da educação financeira
#
# Muitas pessoas não dão importância à educação financeira, mas ela é essencial para o sucesso financeiro. A educação financeira ensina as pessoas a gerenciar o dinheiro de forma eficiente e a fazer escolhas financeiras sábias.
#
# As pessoas que têm um bom conhecimento sobre educação financeira tendem a ganhar mais dinheiro, economizar mais dinheiro e ter menos dívidas. A educação financeira também pode ajudar as pessoas a serem mais bem-sucedidas em seus empregos e a ter uma vida financeira mais estável.
#
# A educação financeira é importante para todas as pessoas, independentemente da idade, renda ou situação financeira. Se você quer aprender mais sobre educação financeira, há muitos recursos disponíveis, incluindo livros, sites da internet, programas de televisão e cursos.
#
# Elapsed Time: 10.05 seconds
